### PR TITLE
docs(rustc): Improve discoverable of Cargo docs

### PR DESCRIPTION
**Pull request by epage**
In preparing Cargo's blog post for 1.80, I tried to find the documentation for the lint configuration and I couldn't.  The link is only visible from the lint itself, which isn't where I started, and the side bar, which was collapsed for me.

The first place I went was the docs for `unexpected_cfgs` because this is configuration for that lint.  If using lint configuration were a one off, I could see skipping it here.  However, when we discussed this with at least one T-compiler member, there was interest in using this for other lints in the future.  To that end, it seems like we should be exposing this with the lint itself.

The second place I checked was the `check-cfg` documentation.  This now has a call out for the sub-page.

### CODE REVIEW

File: src/doc/rustc/src/check-cfg.md [hunk 1]
```diff
@@ -14,6 +14,9 @@ whether to check conditions and how to check them.
 > **Note:** No implicit expectation is added when using `--cfg`. Users are expected to
 pass all expected names and values using the _check cfg specification_.
 
+For interacing with this through Cargo,
```
Comment (lqd) [inline src/doc/rustc/src/check-cfg.md:17]: Is this interacting, or interfacing, or something else?

File: src/doc/rustc/src/check-cfg.md [hunk 2]
```diff
@@ -14,6 +14,9 @@ whether to check conditions and how to check them.
...
+see [Cargo Specifics - Checking Conditional Configurations](check-cfg/cargo-specifics.md).
```
Comment (Urgau) [inline src/doc/rustc/src/check-cfg.md:18]: Isn't a bit redundant to call out the "Checking Conditional Configurations" part inside the check-cfg page?

Why not?
```suggestion
For interacing with this through Cargo,
see the [Cargo Specifics](check-cfg/cargo-specifics.md) subpage.
```

File: src/doc/rustc/src/check-cfg.md (hunk 1 above)
Comment (Urgau) [inline src/doc/rustc/src/check-cfg.md:17]:
```suggestion
For interacting with this through Cargo,
```

File: compiler/rustc_lint_defs/src/builtin.rs [hunk 3]
```diff
@@ -3256,7 +3256,11 @@ declare_lint! {
     /// See the [Checking Conditional Configurations][check-cfg] section for more
     /// details.
     ///
+    /// See the [Cargo Specifics][unexpected_cfgs_lint_config] for configuring this lint in
+    /// `Cargo.toml`.
```
Comment (Urgau) [inline compiler/rustc_lint_defs/src/builtin.rs:3260]: What's the reason for specifically calling out `Cargo.toml` here ? and not simply using "Cargo"?

File: src/doc/rustc/src/check-cfg.md (hunk 1 above)
Comment (epage) [reply src/doc/rustc/src/check-cfg.md]: interacting

File: src/doc/rustc/src/check-cfg.md (hunk 2 above)
Comment (epage) [reply src/doc/rustc/src/check-cfg.md]: Tweaked the wording

File: compiler/rustc_lint_defs/src/builtin.rs (hunk 3 above)
Comment (epage) [reply compiler/rustc_lint_defs/src/builtin.rs]: If this were about check-cfg, there'd be less of a reason for this to belong here.  I'm specifically wanting to highlight lint configuration which also aligns with my motivation for why this belongs in this section as we generally should call out lint configuration here.

**Review by Urgau — APPROVED** [review]

**Comment by Urgau** [comment]
Thanks! Looks good to me.

@bors r+ rollup
r? @Urgau

**Pull request merged**